fix(eslint-plugin): [consistent-type-imports] crash when using both default and namespace in one import

ota-meshi opened the pull request:
Fixes #2775

bradzacher commented on packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts:
```diff
@@ -1085,5 +1090,68 @@ import  { Value4 } from 'default_and_named_import';
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type Default from 'module';
+import  * as Rest from 'module';
```
Hard to notice, but this is being fixed with a double space: `import  *`.
We probably don't want to be doing that if we can avoid it.

bradzacher commented on packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts:
```diff
@@ -1085,5 +1090,68 @@ import  { Value4 } from 'default_and_named_import';
+        const a: Rest.A = '';
+      `,
+      options: [{ prefer: 'type-imports' }],
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type * as Rest from 'module';
+import Default from 'module';
+        const a: Rest.A = '';
+      `,
```
I'd consider just dedenting the entire block to 0 (both code and output) so it matches instead of disabling the lint rule
```suggestion
      output: noFormat`
import type * as Rest from 'module';
import Default from 'module';
const a: Rest.A = '';
      `,
```

bradzacher commented on packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts:
```diff
@@ -1085,5 +1090,68 @@ import  { Value4 } from 'default_and_named_import';
+        const a: Default = '';
+      `,
+      options: [{ prefer: 'type-imports' }],
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type Default from 'module';
+import  * as Rest from 'module';
+        const a: Default = '';
+      `,
```
```suggestion
      output: noFormat`
import type Default from 'module';
import  * as Rest from 'module';
const a: Default = '';
      `,
```

bradzacher commented on packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts:
```diff
@@ -1085,5 +1090,68 @@ import  { Value4 } from 'default_and_named_import';
+        const b: Rest.A = '';
+      `,
+      options: [{ prefer: 'type-imports' }],
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type * as Rest from 'module';
+import type Default from 'module';
+        const a: Default = '';
+        const b: Rest.A = '';
+      `,
```
```suggestion
      output: noFormat`
import type * as Rest from 'module';
import type Default from 'module';
const a: Default = '';
const b: Rest.A = '';
      `,
```

bradzacher reviewed (changes requested):
thanks for hopping in to fix this!

ota-meshi commented:
Thank you for checking! I changed this PR.

I didn't know that 0 indentation would eliminate the format error. Thank you for telling me about this. I also fixed some related pre-existing test cases.

bradzacher reviewed (approved):
thanks for working on this!
